snapcraft/cli/store.py

```diff
@@ -542,11 +542,71 @@ def list_revisions(snap_name, arch):
         snapcraft list-revisions my-snap --arch armhf
         snapcraft revisions my-snap
     """
-    snapcraft.revisions(snap_name, arch)
+    releases = StoreClientCLI().get_snap_releases(snap_name=snap_name)
+
+    def get_channels_for_revision(revision: int) -> List[str]:
```

Comment:
Is the scanning for "seen" just to indicate the first one found is a "current" channel?  It feels a bit error prone if so, but I'm probably misunderstanding.

If it is, could we maybe we use channel map to determine "current" status? e.g.:

active_channels_for_revision = [cm.channel for cm in channel_map if cm.revision == revision]
all_channels_for_revision = [release.channel for release in releases.releases if release.revision == revision]

Reply:
it is the current channel for the entire list of revisions